ast.ts

```diff
@@ -27,19 +30,23 @@ export type Stmt<A> =
   | {  a?: A, tag: "index-assign", obj: Expr<A>, index: Expr<A>, value: Expr<A> }
   | {  a?: A, tag: "if", cond: Expr<A>, thn: Array<Stmt<A>>, els: Array<Stmt<A>> }
   | {  a?: A, tag: "while", cond: Expr<A>, body: Array<Stmt<A>> }
+  | {  a?: A, tag: "nonlocal", name: string }
 
+export type Lambda<A> = {  a?: A, tag: "lambda", params: Array<string>, type: Callable, expr: Expr<A> };
 export type Expr<A> =
     {  a?: A, tag: "literal", value: Literal }
   | {  a?: A, tag: "id", name: string }
   | {  a?: A, tag: "binop", op: BinOp, left: Expr<A>, right: Expr<A>}
   | {  a?: A, tag: "uniop", op: UniOp, expr: Expr<A> }
   | {  a?: A, tag: "builtin1", name: string, arg: Expr<A> }
   | {  a?: A, tag: "builtin2", name: string, left: Expr<A>, right: Expr<A>}
-  | {  a?: A, tag: "call", name: string, arguments: Array<Expr<A>> } 
+  | {  a?: A, tag: "call", fn: Expr<A>, arguments: Array<Expr<A>> } 
```

Comment:
Oof merge conflict alert. I wonder if there's a way to split this into a string version and an expression version? Definitely at least the folks working on calling conventions will conflict with this, probably others.

Reply:
We've been talking to the calling conventions team and they say they can work with this version of the AST. We'll keep in touch with them and see if anything changes.

We've also looked through the other PRs and we don't think this change will interfere with any of the other groups severely.